### PR TITLE
Fix SyncGranule checksum memory issue

### PR DESCRIPTION
**Pull request by yjpa7145**
**Summary:** Fix a bug where S3 objects were loaded into memory before calculating their checksum

Addresses [CUMULUS-2039: reduce memory footprint of checksum verification](https://bugs.earthdata.nasa.gov/browse/CUMULUS-2039)

## Changes

* Remove the `getS3ObjectReadStreamAsync` function, which was reading the S3 object into memory before returning a readable stream for it
* Update the `calculateS3ObjectChecksum` and `validateS3ObjectChecksum` functions to use `getObjectReadStream`, which does not read the entire object before returning a read stream
* Add a test which invokes the `SyncGranule` lambda function and calculates the checksum of a 1 GB file. Verified that it failed before this fix and passed afterward.

## PR Checklist

- [x] Update CHANGELOG
- ~[ ] Unit tests~
- [x] Adhoc testing
- [x] Integration tests

### CODE REVIEW

File: example/spec/parallel/SyncGranuleTask/ChecksumOneGigFileSpec.js [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+      granuleId = randomId('granule-id-');
+
+      const FunctionName = `${prefix}-SyncGranule`;
+
+      const functionConfig = await lambda().getFunctionConfiguration({
+        FunctionName,
+      }).promise();
+
+      const Payload = JSON.stringify({
+        cma: {
```
Comment (Jkovarik) [inline example/spec/parallel/SyncGranuleTask/ChecksumOneGigFileSpec.js:57]: Not for this PR, but this feels like the sort of thing it'd be nice to boilerplate and store with the component,  particularly if we start testing individual lambdas (which we should be).

File: example/fake-provider-cf.yml [hunk 2]
```diff
@@ -199,7 +199,15 @@ Resources:
             #!/bin/sh
 
             dd if=/dev/zero bs=300M count=1 | aws s3 cp - s3://${S3ProviderBucket}/300M.dat
+            dd if=/dev/zero bs=512M count=2 > ~/1G.dat
             dd if=/dev/zero bs=512M count=22 | aws s3 cp - s3://${S3ProviderBucket}/11G.dat
+
+            sha256sum ~/1G.dat | awk '{printf "%s",$1}' > ~/1G.dat.sha256
+
+            aws s3 cp ~/1G.dat s3://${S3ProviderBucket}/1G.dat
```
Comment (Jkovarik) [inline example/fake-provider-cf.yml:207]: Was 1G chosen because the current memory size is 1024?

File: packages/aws-client/src/S3.ts [hunk 3]
```diff
@@ -814,7 +798,7 @@ export const calculateS3ObjectChecksum = deprecate(
     }
   ) => {
     const { algorithm, bucket, key, options } = params;
-    const fileStream = await getS3ObjectReadStreamAsync(bucket, key);
+    const fileStream = getObjectReadStream({ s3: s3(), bucket, key });
```
Comment (Jkovarik) [inline packages/aws-client/src/S3.ts:801]: 👍

File: example/fake-provider-cf.yml (hunk 2 above)
Comment (yjpa7145) [reply example/fake-provider-cf.yml:207]: Yeeeeeesss. I thought the memory was 300MB. I did confirm that this test failed before the changes and passed after, so I think it's still valid.

**Pull request merged**